src/AppBundle/Service/ProductSelectionService.php

```diff
@@ -0,0 +1,162 @@
+ * Time: 10.56
+ */
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\Order;
+use AppBundle\Entity\Product;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+class ProductSelectionService
+{
+    private $em;
+    private $container;
+    private $facebook;
+    private $hoursToRevealSecret;
+
+
+    /**
+     * ProductSelectionService constructor.
+     *
+     * @param EntityManager $em
+     * @param ContainerInterface $container
+     * @param Session $session
+     * @param FacebookInfoService $facebook
+     */
+    public function __construct(EntityManager $em, ContainerInterface $container, FacebookInfoService $facebook)
+    {
+        $this->em = $em;
+        $this->container = $container;
+        $this->facebook =  $facebook;
+        $this->hoursToRevealSecret = $container->getParameter('secret_reveal_time') * 24;
+    }
+
+    /**
+     * @param integer $userId
+     * @param integer $daysToRevealSecret
+     *
+     * @return null|array
+     */
+    private function getUserUnusedProducts($userId)
+    {
+        $validOrderDate = new \DateTime();
+        $validOrderDate->modify('-' . $this->hoursToRevealSecret . ' hours');
+        $revealedProducts = $this->em->getRepository(Order::class)->getUserRevealedProducts(
+            $userId,
+            $validOrderDate
+        );
+
+        $validProductDate = new \DateTime();
+        $validProductDate->modify('+' . $this->hoursToRevealSecret . ' hours');
+        $newProducts = $this->em->getRepository(Product::class)->getUniqueProducts($revealedProducts, $validProductDate);
+
+        if (empty($newProducts)) {
+            return null;
+        } else {
+            return $newProducts;
+        }
+    }
+
+    public function selectProperProduct($userId)
+    {
+        $unusedProducts = $this->getUserUnusedProducts($userId);
+
+        if ($unusedProducts) {
+            $userLikes = $this->facebook->getUserDataByReference('likes');
+            $userEvents = $this->facebook->getUserDataByReference('events?fields=name');
+
+            $matchInLikes = $this->matchesInArrays($unusedProducts, $userLikes, 'name');
+            $matchInEvents = $this->matchesInArrays($unusedProducts, $userEvents, 'name');
+            $productsMatch = array_merge($matchInLikes, $matchInEvents);
+
+
+            if (!empty($productsMatch)) {
+                $personInfo = $this->facebook->getPersonInfo('gender,birthday');
+                $age = $this->getUserAge($personInfo['birthday'])->y;
+                $gender = $personInfo['gender'];
+                $productsByGender = [];
+
+                foreach ($productsMatch as $product) {
+                    if (($product['gender'] == "unisex") || ($product['gender'] == $gender)) {
+                        array_push($productsByGender, $product);
+                    }
+                }
+
+                if (!empty($productsByGender)) {
```

Comment:
You should refactor this huge if's block.. Split it to separate methods or smth... It's just an offer ;)